src/dsync/dsync.c

```diff
@@ -1591,8 +1592,50 @@ static int dsync_strmap_compare(
-        strmap_setf(metadata_refresh, "%llu=%llu", src_index, dst_index);
+         * refresh list, only include those that have different metadata. */
+        uid_t src_uid = (uid_t) mfu_flist_file_get_uid(src_list, src_index);
+        uid_t dst_uid = (uid_t) mfu_flist_file_get_uid(dst_list, dst_index);
+        gid_t src_gid = (gid_t) mfu_flist_file_get_gid(src_list, src_index);
+        gid_t dst_gid = (gid_t) mfu_flist_file_get_gid(dst_list, dst_index);
+        uint64_t src_perm = mfu_flist_file_get_perm(src_list, src_index);
+        uint64_t dst_perm = mfu_flist_file_get_perm(dst_list, dst_index);
+        uint64_t src_atime      = mfu_flist_file_get_atime(src_list, src_index);
+        uint64_t src_atime_nsec = mfu_flist_file_get_atime_nsec(src_list, src_index);
+        uint64_t dst_atime      = mfu_flist_file_get_atime(dst_list, dst_index);
+        uint64_t dst_atime_nsec = mfu_flist_file_get_atime_nsec(dst_list, dst_index);
+        uint64_t src_mtime      = mfu_flist_file_get_mtime(src_list, src_index);
+        uint64_t src_mtime_nsec = mfu_flist_file_get_mtime_nsec(src_list, src_index);
+        uint64_t dst_mtime      = mfu_flist_file_get_mtime(dst_list, dst_index);
+        uint64_t dst_mtime_nsec = mfu_flist_file_get_mtime_nsec(dst_list, dst_index);
+        if ((src_uid != dst_uid) || (src_gid != dst_gid) || (src_perm != dst_perm) ||
+            (src_atime != dst_atime) || (src_atime_nsec != dst_atime_nsec) ||
+            (src_mtime != dst_mtime) || (src_mtime_nsec != dst_mtime_nsec))
+        {
+            /* update to say user name of the files were found to be different */
+            if ((src_uid != dst_uid)) {
+                dsync_strmap_item_update(src_map, key, DCMPF_UID, DCMPS_DIFFER);
+                dsync_strmap_item_update(dst_map, key, DCMPF_UID, DCMPS_DIFFER);
+            }
+            if ((src_gid != dst_gid)) {
+                dsync_strmap_item_update(src_map, key, DCMPF_GID, DCMPS_DIFFER);
+                dsync_strmap_item_update(dst_map, key, DCMPF_GID, DCMPS_DIFFER);
+            }
+            if ((src_perm != dst_perm)) {
+                dsync_strmap_item_update(src_map, key, DCMPF_PERM, DCMPS_DIFFER);
+                dsync_strmap_item_update(dst_map, key, DCMPF_PERM, DCMPS_DIFFER);
+            }
+            if ((src_atime != dst_atime) || (src_atime_nsec != dst_atime_nsec)) {
+                dsync_strmap_item_update(src_map, key, DCMPF_ATIME, DCMPS_DIFFER);
+                dsync_strmap_item_update(dst_map, key, DCMPF_ATIME, DCMPS_DIFFER);
+            }
+            if ((src_mtime != dst_mtime) || (src_mtime_nsec != dst_mtime_nsec)) {
+                dsync_strmap_item_update(src_map, key, DCMPF_MTIME, DCMPS_DIFFER);
+                dsync_strmap_item_update(dst_map, key, DCMPF_MTIME, DCMPS_DIFFER);
+            }
+
+            /* mark file as metadata update required */
+            strmap_setf(metadata_refresh, "%llu=%llu", src_index, dst_index);
+        } 
```

Comment:
What I'm thinking is that rather than duplicating all of the code that checks for differences in metadata, we could modify the ``mfu_flist_file_sync_meta`` function to add a new test/change flag parameter.  If the test flag is set, it would not actually make any changes, but it would return 1 if it finds that it needs to make a change and returns 0 otherwise.  Then we could simplify this code to be something like:
```suggestion
         * refresh list, only include those that have different metadata. */
        if (mfu_flist_file_sync_meta(src_list, src_index, dst_list, dst_index, mfu_dst_file, 1) == 1) {
            /* mark file as metadata update required */
            strmap_setf(metadata_refresh, "%llu=%llu", src_index, dst_index);
        } 
```